velox/functions/prestosql/aggregates/tests/VarianceAggregationTest.cpp

```diff
@@ -201,153 +155,79 @@ TEST_F(VarianceAggregationTest, variance) {
-    agg = PlanBuilder()
-              .values(vectors)
-              .project({"c0 % 10", "c1", "c2", "c3", "c4", "c5"})
-              .partialAggregation(
-                  {"p0"},
-                  {GEN_AGG("c1"),
-                   GEN_AGG("c2"),
-                   GEN_AGG("c3"),
-                   GEN_AGG("c4"),
-                   GEN_AGG("c5")})
-              .finalAggregation()
-              .planNode();
     sql = genAggrQuery(
         "SELECT c0 % 10, {0}(c1), {0}(c2), {0}(c3::DOUBLE), {0}(c4), {0}(c5) "
         "FROM tmp GROUP BY 1",
         aggrName);
-    assertQuery(agg, sql);
+    testAggregations(
+        [&](PlanBuilder& builder) {
+          builder.values(vectors).project(
+              {"c0 % 10", "c1", "c2", "c3", "c4", "c5"});
+        },
+        {"p0"},
+        {GEN_AGG("c1"),
+         GEN_AGG("c2"),
+         GEN_AGG("c3"),
+         GEN_AGG("c4"),
+         GEN_AGG("c5")},
+        sql);
 
-    agg = PlanBuilder()
-              .values(vectors)
-              .project({"c0 % 10", "c1", "c2", "c3", "c4", "c5"})
-              .singleAggregation(
-                  {"p0"},
-                  {GEN_AGG("c1"),
-                   GEN_AGG("c2"),
-                   GEN_AGG("c3"),
-                   GEN_AGG("c4"),
-                   GEN_AGG("c5")})
-              .planNode();
-    sql = genAggrQuery(
-        "SELECT c0 % 10, {0}(c1), {0}(c2), {0}(c3::DOUBLE), {0}(c4), {0}(c5) "
-        "FROM tmp GROUP BY 1",
-        aggrName);
-    assertQuery(agg, sql);
-
-    agg = PlanBuilder()
-              .values(vectors)
-              .project({"c0 % 10", "c1", "c2", "c3", "c4", "c5"})
-              .partialAggregation(
-                  {"p0"},
-                  {GEN_AGG("c1"),
-                   GEN_AGG("c2"),
-                   GEN_AGG("c3"),
-                   GEN_AGG("c4"),
-                   GEN_AGG("c5")})
-              .intermediateAggregation()
-              .finalAggregation()
-              .planNode();
+    // Group by; no input
     sql = genAggrQuery(
-        "SELECT c0 % 10, {0}(c1), {0}(c2), {0}(c3::DOUBLE), {0}(c4), {0}(c5) "
-        "FROM tmp GROUP BY 1",
+        "SELECT c0 % 10, {0}(c1) FROM tmp WHERE c0 % 2 = 5 GROUP BY 1",
         aggrName);
-    assertQuery(agg, sql);
-
-    // Group by; no input
-    agg = PlanBuilder()
-              .values(vectors)
-              .project({"c0 % 10 AS c0_mod_10", "c1"})
-              .filter("c0_mod_10 > 10")
-              .partialAggregation({"c0_mod_10"}, {GEN_AGG("c1")})
-              .finalAggregation()
-              .planNode();
-    assertQueryReturnsEmptyResult(agg);
+    testAggregations(
+        [&](PlanBuilder& builder) {
+          builder.values(vectors)
+              .filter("c0 % 2 = 5")
+              .project({"c0 % 10", "c1"});
+        },
+        {"p0"},
+        {GEN_AGG("c1")},
+        sql);
 
     // Group by over filter
-    agg = PlanBuilder()
-              .values(vectors)
-              .filter("c2 % 5 = 3")
```

Comment:
Any reason to modify this filter expression?

Reply:
Nice catch. Let me fix.